mlflow/R/mlflow/R/databricks-utils.R

```diff
@@ -91,3 +91,37 @@ get_databricks_config <- function(profile) {
   }
   config
 }
+
+mlflow_get_run_context.mlflow_databricks_client <- function(client, source_name, source_version,
+                                                            source_type, ...) {
+  if (exists("databricks_get_notebook_info")) {
+    notebook_info <- do.call("databricks_get_notebook_info", list())
+    if (!is.na(notebook_info$id) && !is.na(notebook_info$path)) {
+      tags <- list()
+      tags[[MLFLOW_DATABRICKS_TAGS$MLFLOW_DATABRICKS_NOTEBOOK_ID]] <- notebook_info$id
+      tags[[MLFLOW_DATABRICKS_TAGS$MLFLOW_DATABRICKS_NOTEBOOK_PATH]] <- notebook_info$path
+      tags[[MLFLOW_DATABRICKS_TAGS$MLFLOW_DATABRICKS_WEBAPP_URL]] <- notebook_info$webapp_url
+      list(
+        client = client,
+        source_version = source_version %||% get_source_version(),
+        source_type =  MLFLOW_SOURCE_TYPE$NOTEBOOK,
```

Comment:
One observation entirely orthogonal to this PR, I guess we don't set `source_type` elsewhere in the R client - we might want to consider defaulting `source_type` to `LOCAL` [as the Python client does](https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/fluent.py#L154), since the runs UI renders an icon conditional on source type (i.e. add a `get_source_type` function in `tracking-utils.R` and then invoke it in `mlflow_get_run_context.default` within `tracking-client.R`. Will make a GitHub issue to track it separately

Reply:
That's a good point!